plugins/AuthUI/src/androidMain/kotlin/io/github/temk0/supabase/authui/ui/GoogleButton.kt

```diff
@@ -0,0 +1,74 @@
+package io.github.temk0.supabase.authui.ui
+
+import android.app.Activity
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.IntentSenderRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalContext
+import com.google.android.gms.auth.api.identity.Identity
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.CommonStatusCodes
+import io.github.temk0.supabase.authui.AuthUI
+import io.github.temk0.supabase.authui.AuthUIImpl
+import io.github.temk0.supabase.authui.GoogleLoginConfig
+import io.github.temk0.supabase.authui.getSignInRequest
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+@Composable
+actual fun AuthUI.rememberLoginWithGoogle(
+    onResult: (NativeSignInResult) -> Unit,
+    fallback: () -> Unit
+): NativeSignInState {
+    this as AuthUIImpl
+
+    val state = remember { NativeSignInState() }
+
+    // init signInRequest options
+    if (config.loginConfig == null || config.loginConfig !is GoogleLoginConfig) {
+        fallback.invoke()
+        state.reset()
+        return state
+    }
+
+    val config = config.loginConfig as GoogleLoginConfig?
+    val signInRequest = getSignInRequest(config)
+    val oneTabClient = Identity.getSignInClient(LocalContext.current)
```

Comment:
```suggestion
    // init signInRequest options
    if (config.loginConfig == null || config.loginConfig !is GoogleLoginConfig) {
        fallback.invoke()
        state.reset()
        return state
    }

    val config = config.loginConfig as GoogleLoginConfig?
    val signInRequest = getSignInRequest(config)
    val oneTabClient = Identity.getSignInClient(LocalContext.current)
```
I would also include this in the "LaunchedEffect" block, as you are recreating the sign in request and calling the fallback on every recomposition. You can just declare the android context before that.